cmd/sync-server/main.go

```diff
@@ -0,0 +1,261 @@
+package main
+
+import (
+	"github.com/wptide/pkg/sync"
+	"github.com/wptide/pkg/message"
+	"github.com/wptide/pkg/wporg"
+	"time"
+	"fmt"
+	"strings"
+	"github.com/wptide/pkg/env"
+	"log"
+	"strconv"
+)
+
+var (
+	// Sync Server config.
+	serverActive    = strings.ToLower(env.GetEnv("SYNC_ACTIVE", "off")) == "on"
+	forcedSync      = strings.ToLower(env.GetEnv("SYNC_FORCE_AUDITS", "no")) == "yes"
+	apiClient       = env.GetEnv("SYNC_DEFAULT_CLIENT", "wporg")
+	auditVisibility = env.GetEnv("SYNC_DEFAULT_VISIBILITY", "public")
+	perPage, _      = strconv.Atoi(env.GetEnv("SYNC_ITEMS_PER_PAGE", "250"))
+	poolSize, _     = strconv.Atoi(env.GetEnv("SYNC_POOL_WORKERS", "250"))
+	browseCategory  = env.GetEnv("SYNC_API_BROWSE_CATEGORY", "updated")
+	poolDelay, _    = strconv.Atoi(env.GetEnv("SYNC_POOL_DELAY", "300"))
+
+	// Allow us to quit the server with a channel. Good for tests.
+	quit = make(chan struct{}, 1)
+
+	// Fetches themes and plugins from the WordPress.org API.
+	client = &wporg.Client{}
+
+	// Checks to see if we need to send an update.
+	checkerDBPath = env.GetEnv("SYNC_DATA", "./db")
+
+	// Init the scribble (flat file) db used for checking currency of results.
+	checker sync.UpdateChecker = scribbleChecker{
+		db: newScribbleChecker(checkerDBPath),
+	}
+
+	// Create the SQS dispatcher.
+	dispatcher = &sqsDispatcher{
+		Queues: map[string]struct {
+			Region   string
+			Key      string
+			Secret   string
+			Queue    string
+			Endpoint string
+			Active   bool
+			Accepts  string // "all" or "themes" or "plugins"
+		}{
+			"primary": {
```

Comment:
I'm thinking that we should have two queues, a PHPCS queue and a Lighthouse queue. What purpose does this configuration serve? We don't have an immediate need for a plugin queue and being explicit about what the queue is makes more sense semantically to me than generic queues for themes and plugins.